fastcrypto-zkp/src/api.rs

```diff
@@ -0,0 +1,30 @@
+use std::io::Cursor;
+
+use ark_bls12_381::Bls12_381;
+use ark_groth16::{Proof, VerifyingKey};
+use ark_relations::r1cs::SynthesisError;
+use ark_serialize::CanonicalDeserialize;
+
+use crate::conversions::BlsFr;
+use crate::verifier::{process_vk_special, verify_with_processed_vk};
+
+/// Verify Groth16 proof using the serialized representation of verifying key, input and proof.
+/// This assumes canoncical ser/de defined in Arkworks, instead of the Zcash bytes representation.
+pub fn verify_groth16_in_bytes(
+    vk_bytes: Vec<u8>,
+    x_bytes: Vec<u8>,
+    proof_bytes: Vec<u8>,
+) -> Result<bool, SynthesisError> {
```

Comment:
should we use FastCryptoError here? (currently the enum is not accessible in fastcrypto-zkp)

Reply:
It should be preferable to have our own controllable fastcrypto error types (this will also make our system backend agnostic, cause now SynthesisError requires an akrworks dependency when outputting errors)

Reply:
I'm up for exposing our own error type as well, but a minimalistic approach should be fine IMHO (i.e. no need to re-implement all of `SynthesisError`).